### PR TITLE
support constructing Panel or Panel4D with scalar data, fixes #8285

### LINKED ISSUE #8285: BUG? Can construct constant Series and DataFrame, but not Panel or Panel4D
Is it accidental that the Series and DataFrame constructors accept a single number for ``data``, but the Panel and Panel4D constructors do not?

```
In [182]: from pandas import Series, DataFrame, Panel, Panel4D

In [183]: Series(1., index=range(2))
Out[183]:
0    1
1    1
dtype: float64

In [184]: DataFrame(1., index=range(2), columns=range(2))
Out[184]:
   0  1
0  1  1
1  1  1

In [185]: Panel(1., items=range(2), major_axis=range(2), minor_axis=range(2))
---------------------------------------------------------------------------
PandasError                               Traceback (most recent call last)
<ipython-input-185-89f6618b848f> in <module>()
----> 1 Panel(1., items=range(2), major_axis=range(2), minor_axis=range(2))

C:\Python34\lib\site-packages\pandas\core\panel.py in __init__(self, data, items, major_axis, minor_axis, copy, dtype)
    133                  copy=False, dtype=None):
    134         self._init_data(data=data, items=items, major_axis=major_axis,
--> 135                         minor_axis=minor_axis, copy=copy, dtype=dtype)
    136
    137     def _init_data(self, data, copy, dtype, **kwargs):

C:\Python34\lib\site-packages\pandas\core\panel.py in _init_data(self, data, copy, dtype, **kwargs)
    161             dtype = None
    162         else:  # pragma: no cover
--> 163             raise PandasError('Panel constructor not properly called!')
    164
    165         NDFrame.__init__(self, mgr, axes=axes, copy=copy, dtype=dtype)

PandasError: Panel constructor not properly called!

In [186]: Panel4D(1., labels=range(2), items=range(2), major_axis=range(2), minor_axis=range(2))
---------------------------------------------------------------------------
PandasError                               Traceback (most recent call last)
<ipython-input-187-8f22d9107a3e> in <module>()
----> 1 Panel4D(1., labels=range(2), items=range(2), major_axis=range(2), minor_axis=range(2))

C:\Python34\lib\site-packages\pandas\core\panel4d.py in panel4d_init(self, data, labels, items, major_axis, minor_axis, copy, dtype)
     37     self._init_data(data=data, labels=labels, items=items,
     38                     major_axis=major_axis, minor_axis=minor_axis,
---> 39                     copy=copy, dtype=dtype)
     40
     41 Panel4D.__init__ = panel4d_init

C:\Python34\lib\site-packages\pandas\core\panel.py in _init_data(self, data, copy, dtype, **kwargs)
    161             dtype = None
    162         else:  # pragma: no cover
--> 163             raise PandasError('Panel constructor not properly called!')
    164
    165         NDFrame.__init__(self, mgr, axes=axes, copy=copy, dtype=dtype)

PandasError: Panel constructor not properly called!
```

### PR DESCRIPTION
**Pull request by mortada**
this should fix https://github.com/pydata/pandas/issues/8285

### CODE REVIEW

File: pandas/tests/test_panel.py [hunk 1]
```diff
@@ -888,6 +888,21 @@ def test_constructor(self):
         wp = Panel(vals, copy=True)
         self.assertIsNot(wp.values, vals)
 
+        # test when scalar data is used to construct a Panel
+        # if dtype is not passed, it should be inferred
+        value_and_dtype = [(1, int), (3.14, float), ('foo', np.object_)]
```
Comment (jreback) [inline pandas/tests/test_panel.py]: add the issue number here

File: pandas/tests/test_panel4d.py [hunk 2]
```diff
@@ -629,6 +629,21 @@ def test_constructor(self):
         panel4d = Panel4D(vals, copy=True)
         self.assertIsNot(panel4d.values, vals)
 
+        # test when scalar data is used to construct a Panel4D
+        # if dtype is not passed, it should be inferred
+        value_and_dtype = [(1, int), (3.14, float), ('foo', np.object_)]
+        for (val, dtype) in value_and_dtype:
```
Comment (jreback) [inline pandas/tests/test_panel4d.py]: same here

**Comment by jreback** [comment]
ok, pls add a release note (this can go in 0.16.0)

File: pandas/core/panel.py [hunk 3]
```diff
@@ -165,6 +165,13 @@ def _init_data(self, data, copy, dtype, **kwargs):
             mgr = self._init_matrix(data, passed_axes, dtype=dtype, copy=copy)
             copy = False
             dtype = None
+        elif lib.isscalar(data) and all(x is not None for x in passed_axes):
+            if dtype is None:
+                dtype, data = _infer_dtype_from_scalar(data)
+            values = np.empty([len(x) for x in passed_axes], dtype=dtype)
+            values.fill(data)
+            mgr = self._init_matrix(values, passed_axes, dtype=dtype, copy=copy)
```
Comment (jreback) [inline pandas/core/panel.py]: I think you need to make ``copy= False`` before you init the matrix (so it doesn't copy it again).

**Comment by mortada** [comment]
@jreback thanks it's fixed now

**Comment by jreback** [comment]
@mortada thanks!

**Pull request merged**